vcd/resource_vcd_catalog_item.go

```diff
@@ -135,76 +133,19 @@ func resourceVcdCatalogItemCreate(ctx context.Context, d *schema.ResourceData, m
 	return resourceVcdCatalogItemRead(ctx, d, meta)
 }
 
-func uploadFile(d *schema.ResourceData, catalog *govcd.Catalog, itemName string) diag.Diagnostics {
```

Comment:
Is the main reason of getting rid of this function that we cannot print upload progress to the screen anymore? Are we printing it to the log instead?

Reply:
What happened here is several things:

- I moved these auxiliary functions to the `resource_vcd_catalog_vapp_template.go` file, just to be prepared if we ever deprecate `resource_catalog_item`.

- I then renamed it to `uploadOvaFromFilePath` as we had already an `uploadFromUrl`, so it fits better to what it's doing.

- I refactored also `uploadOvaFromFilePath` and `finishHandlingTask` as they were both doing the same treatment to the `show_upload_progress ` attribute. So now that progress handling is now done only in `finishHandlingTask` to avoid duplication.

- As you noticed, `finishHandlingTask` is only printing the progress for `vcd_catalog_item`. The reason is what you said, this attribute was introduced in a period of time where printing to screen was OK, but as we can't do it anymore it doesn't make much sense (the attribute is not even in the new `vcd_catalog_vapp_template` resource)